controllers/scopeinstance_controller.go

```diff
@@ -143,12 +192,29 @@ func (r *ScopeInstanceReconciler) Reconcile(ctx context.Context, req ctrl.Reques
 	}
 
 	if err := r.deleteBindings(ctx, listOption, listOptions); err != nil {
+		r.updateScopeTemplateCondition(ctx, in, metav1.Condition{
+			Type:               "Succeeded",
+			Status:             metav1.ConditionFalse,
+			ObservedGeneration: in.Generation,
+			LastTransitionTime: metav1.Now(),
+			Reason:             "RoleBindingDeleteFailure",
+			Message:            fmt.Sprintf("failed to delete RoleBindings associated with ScopeInstance `%s`: %s", in.Name, err),
+		})
 		log.Log.Info("Error in deleting Role Bindings", "error", err)
 		return ctrl.Result{}, err
 	}
 
 	log.Log.Info("No ScopeInstance error")
 
+	r.updateScopeTemplateCondition(ctx, in, metav1.Condition{
+		Type:               "Succeeded",
+		Status:             metav1.ConditionTrue,
+		ObservedGeneration: st.Generation,
+		LastTransitionTime: metav1.Now(),
+		Reason:             "ScopeInstanceReconcileSuccess",
+		Message:            fmt.Sprintf("ScopeInstance `%s` successfully reconciled", in.Name),
```

Comment:
A few nits/suggestions:
1. Define constants for all types and reasons. These are essentially part of the API, so put them in the file that defines the `ScopeTemplate`
2. No need to set `LastTransitionTime` manually. `meta.SetStatusCondition` handles that complexity for you. There's specific logic around exactly _when_ to change it, which is only when the status for a given condition changes.
3. Be _very_ judicious with Types and Reasons. Again, think of these like APIs. You can't just add and remove them willy nilly because you might have other consumers of your API building logic based on these. Rubric for new types and reasons is generally: is there a use case that requires distinguishing these things. If not, keep the types/reasons fairly general.
```suggestion
		Type:               operatorsv1.TypeSucceeded,
		Status:             metav1.ConditionTrue,
		ObservedGeneration: st.Generation,
		Reason:             operatorsv1.ScopeInstanceReconcileSuccess,
		Message:            fmt.Sprintf("ScopeInstance `%s` successfully reconciled", in.Name),
```

Reply:
So taking another look at the rukpak implementation cleared things up a bit more for me. I went ahead and updated the Type and Reasons to be minimal while still covering what I think are the valid use cases in commit https://github.com/awgreene/oria-operator/pull/21/commits/01d4b49156e5523bab706fd1c4c06ee6d1fa057c .

I felt it was good to keep a reason of `ReasonScopeTemplateNotFound` as someone may want to configure something that would create a `ScopeInstance` and if the Reason for the `ScopeInstance` not being reconciled properly is due to the `ScopeTemplate` it references missing - create it. I removed all the other reasons and replaced them with a more generic `ReasonScopingFailed` as the rest of them I'm not sure there could be any reasonable actions taken by anyone other than a cluster admin (i.e not having the proper permissions for creating/deleting cluster roles).

Let me know what you think!